apps/cli/src/code-generator.ts

```diff
@@ -0,0 +1,146 @@
+import { formatFiles, generateFiles } from '@nx/devkit'
+import { printChanges } from 'nx/src/command-line/generate/generate'
+import { FsTree, flushChanges } from 'nx/src/generators/tree'
+import * as path from 'path'
+import prompts from 'prompts'
+import {
+    getDotNetNameSpace,
+    toDotNetType,
+    toPascalCase,
+} from './templates/functions'
+
+export type TemplateType = 'dotnet-api'
+
+export async function CodeGenerator(options: {
+    templateType: TemplateType
+    outputPath: string
+    organizationName: string
+    featureBoardKey?: string
+    featureBoardBearerToken?: string
+    featureBoardprojectName?: string
+
+    dryRun: boolean
+    quiet: boolean
+    verbose: boolean
+    interactive: boolean
+}): Promise<void> {
+    const features = await getFeatures(options)
+
+    const templateSpecificOptions: any = {}
+
+    switch (options.templateType) {
+        case 'dotnet-api':
+            templateSpecificOptions.namespace = await getDotNetNameSpace(
+                options.outputPath,
+            )
+    }
+
+    const tree = new FsTree(options.outputPath, options.verbose)
+    generateFiles(
+        tree,
+        path.join(__dirname, 'templates', options.templateType),
+        './',
+        {
+            toPascalCase,
+            toDotNetType,
+            ...templateSpecificOptions,
+            features: features,
+        },
+    )
+    await formatFiles(tree)
+
+    SaveChanges(tree, options)
+}
+
+export default CodeGenerator
```

Comment:
If we could not use default exports and also use a lower case `c`, as capital denotes a class/constructor not a function by convention

Reply:
Fixed